### ISSUE TITLE
Fail to export genesis from zero-height

### ISSUE THREAD

**Issue by ewagmig** [CosmosContracts/juno#778]
#### Specifications

Juno version: `v16.0.0-4-gb34af55`
OS & Version: OSX
Commit hash: b34af555ba5918f1aec0aa8c58df7ed89bb653aa

#### Expected behavior
The command export --for-zero-height is used to export the genesis file from the genesis time when the chain launched.

#### Actual behavior
 It failed due to an invalid validator address retrieval panic error.

#### Steps to reproduce the behavior
1. run the node via `make test-node`
2. stop the node after the node runs up normally to reach a height greater than zero
3. export the genesis file with command `junod export --for-zero-height`
4. the process exit with error panic:"panic: expected validator, not found" 

#### Backtrace

````
$junod export --for-zero-height --home "/Users/mingwang/.juno1"
4:05PM INF asserting crisis invariants inv=1/12 module=x/crisis name=bank/nonnegative-outstanding
4:05PM INF asserting crisis invariants inv=2/12 module=x/crisis name=bank/total-supply
4:05PM INF asserting crisis invariants inv=3/12 module=x/crisis name=transfer/total-escrow-per-denom
4:05PM INF asserting crisis invariants inv=4/12 module=x/crisis name=gov/module-account
4:05PM INF asserting crisis invariants inv=5/12 module=x/crisis name=distribution/nonnegative-outstanding
4:05PM INF asserting crisis invariants inv=6/12 module=x/crisis name=distribution/can-withdraw
4:05PM INF asserting crisis invariants inv=7/12 module=x/crisis name=distribution/reference-count
4:05PM INF asserting crisis invariants inv=8/12 module=x/crisis name=distribution/module-account
4:05PM INF asserting crisis invariants inv=9/12 module=x/crisis name=staking/module-accounts
4:05PM INF asserting crisis invariants inv=10/12 module=x/crisis name=staking/nonnegative-power
4:05PM INF asserting crisis invariants inv=11/12 module=x/crisis name=staking/positive-delegation
4:05PM INF asserting crisis invariants inv=12/12 module=x/crisis name=staking/delegator-shares
4:05PM INF asserted all invariants duration=3.389292 height=4 module=x/crisis
panic: expected validator, not found

goroutine 1 [running]:
github.com/CosmosContracts/juno/v17/app.(*App).prepForZeroHeightGenesis(_, {{0x105846670, 0x140000580b8}, {0x10585b480, 0x14002468140}, {{0x0, 0x0}, {0x0, 0x0}, 0x4, ...}, ...}, ...)
        github.com/CosmosContracts/juno/v17/app/export.go:158 +0xf38
github.com/CosmosContracts/juno/v17/app.(*App).ExportAppStateAndValidators(0x14000ffb000, 0x1, {0x106d9a7a8, 0x0, 0x0}, {0x106d9a7a8, 0x0, 0x0})
        github.com/CosmosContracts/juno/v17/app/export.go:27 +0x118
github.com/CosmosContracts/juno/v17/cmd/junod/cmd.appCreator.appExport({{{0x105851ab0, 0x140001e0d20}, {0x1058665a0, 0x14000970050}, {0x105857f00, 0x14000aa3bc0}, 0x140004a8558}}, {0x105847be0, 0x1400117f930}, {0x10585b990, ...}, ...)
        github.com/CosmosContracts/juno/v17/cmd/junod/cmd/root.go:350 +0x144
github.com/cosmos/cosmos-sdk/server.ExportCmd.func1(0x14000166c00, {0x14000ea4180?, 0x0?, 0x3?})
        github.com/cosmos/cosmos-sdk@v0.47.4/server/export.go:73 +0x338
github.com/spf13/cobra.(*Command).execute(0x14000166c00, {0x14000ea40f0, 0x3, 0x3})
        github.com/spf13/cobra@v1.7.0/command.go:940 +0x5c4
github.com/spf13/cobra.(*Command).ExecuteC(0x1400124a300)
        github.com/spf13/cobra@v1.7.0/command.go:1068 +0x340
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.7.0/command.go:992
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        github.com/spf13/cobra@v1.7.0/command.go:985
github.com/cosmos/cosmos-sdk/server/cmd.Execute(0x140000021a0?, {0x104627212, 0x5}, {0x14000054618, 0x15})
        github.com/cosmos/cosmos-sdk@v0.47.4/server/cmd/execute.go:32 +0x13c
main.main()
        github.com/CosmosContracts/juno/v17/cmd/junod/main.go:18 +0x44
````
````

When submitting logs: please submit them as text and not screenshots.

**Comment by Reecepbcups** [comment]
@ewagmig Hey there, this seems to be off the main branch which is unreleased (v17).

please use the v16.0.0 tag for mainnet & testnet. 

---

did make test-node run though and start up just fine on your machine?

**Comment by ewagmig** [comment]
Hey @Reecepbcups, the issue existed in the release of v16.0.0 as well. 
This issue was first introduced from `ignite CLI` or `starport` when scaffolding the chain. Where the root cause lies in the incorrect validator address fetch via the `addr := sdk.ValAddress(iter.Key()[1:])`, which will lead to a panic error when this command is executed.
Note that the validator address is fetched via `iter.Key()[1:]`, which excludes the `ValidatorKey`, but it still includes the prefix, the length of the address. This incomplete path leads to the error: `expected validator, not found`   

Please see the log as below:
```
$ junod version 
v16.0.0

# mingwang @ Mings-MacBook-Pro in ~/codeme/src/github.com/CosmosContracts/juno on git:v16.0.0 x [23:04:39] 
$ junod export --for-zero-height --home "/Users/mingwang/.juno1" 
11:05PM INF asserting crisis invariants inv=1/12 module=x/crisis name=distribution/nonnegative-outstanding
11:05PM INF asserting crisis invariants inv=2/12 module=x/crisis name=distribution/can-withdraw
11:05PM INF asserting crisis invariants inv=3/12 module=x/crisis name=distribution/reference-count
11:05PM INF asserting crisis invariants inv=4/12 module=x/crisis name=distribution/module-account
11:05PM INF asserting crisis invariants inv=5/12 module=x/crisis name=gov/module-account
11:05PM INF asserting crisis invariants inv=6/12 module=x/crisis name=staking/module-accounts
11:05PM INF asserting crisis invariants inv=7/12 module=x/crisis name=staking/nonnegative-power
11:05PM INF asserting crisis invariants inv=8/12 module=x/crisis name=staking/positive-delegation
11:05PM INF asserting crisis invariants inv=9/12 module=x/crisis name=staking/delegator-shares
11:05PM INF asserting crisis invariants inv=10/12 module=x/crisis name=transfer/total-escrow-per-denom
11:05PM INF asserting crisis invariants inv=11/12 module=x/crisis name=bank/nonnegative-outstanding
11:05PM INF asserting crisis invariants inv=12/12 module=x/crisis name=bank/total-supply
11:05PM INF asserted all invariants duration=5.454292 height=5 module=x/crisis
panic: expected validator, not found

goroutine 1 [running]:
github.com/CosmosContracts/juno/v16/app.(*App).prepForZeroHeightGenesis(_, {{0x1054fa210, 0x1400019c008}, {0x10550ee40, 0x1400237b8c0}, {{0x0, 0x0}, {0x0, 0x0}, 0x5, ...}, ...}, ...)
        github.com/CosmosContracts/juno/v16/app/export.go:158 +0xf38
github.com/CosmosContracts/juno/v16/app.(*App).ExportAppStateAndValidators(0x140008fd000, 0x1, {0x106a41c78, 0x0, 0x0}, {0x106a41c78, 0x0, 0x0})
        github.com/CosmosContracts/juno/v16/app/export.go:27 +0x118
github.com/CosmosContracts/juno/v16/cmd/junod/cmd.appCreator.appExport({{{0x105505530, 0x140000acf80}, {0x105519f60, 0x14001033730}, {0x10550b920, 0x140006e3640}, 0x14000198ba8}}, {0x1054fb7b8, 0x1400106f320}, {0x10550f350, ...}, ...)
        github.com/CosmosContracts/juno/v16/cmd/junod/cmd/root.go:350 +0x144
github.com/cosmos/cosmos-sdk/server.ExportCmd.func1(0x1400130b500, {0x14000eb1e90?, 0x0?, 0x3?})
        github.com/cosmos/cosmos-sdk@v0.47.3/server/export.go:73 +0x338
github.com/spf13/cobra.(*Command).execute(0x1400130b500, {0x14000eb1e00, 0x3, 0x3})
        github.com/spf13/cobra@v1.7.0/command.go:940 +0x5c4
github.com/spf13/cobra.(*Command).ExecuteC(0x140012cd500)
        github.com/spf13/cobra@v1.7.0/command.go:1068 +0x340
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.7.0/command.go:992
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        github.com/spf13/cobra@v1.7.0/command.go:985
github.com/cosmos/cosmos-sdk/server/cmd.Execute(0x140000021a0?, {0x1042e0fc6, 0x5}, {0x140001a87e0, 0x15})
        github.com/cosmos/cosmos-sdk@v0.47.3/server/cmd/execute.go:32 +0x13c
main.main()
        github.com/CosmosContracts/juno/v16/cmd/junod/main.go:18 +0x44

```

**Comment by Reecepbcups** [comment]
@ewagmig Thanks for the report! I have got the PRs to patch this both for Main (v17) and v16 branch.

**Comment by Reecepbcups** [comment]
@ewagmig Can you please confirm that the `v16.0.1-alpha.1` tag now works as expected for you? (main branch v17 has also had the patch)

**Comment by CertiK4Audit** [comment]
@Reecepbcups It can be confirmed the patched works well in `v16.0.1-alpha.1` tag now

```
$ junod version                                                 
v16.0.1-alpha.1

# mingwang @ Mings-MacBook-Pro in ~/codeme/src/github.com/CosmosContracts/juno on git:v16.0.1-alpha.1 x [11:48:52] 
$ junod export --for-zero-height --home "/Users/mingwang/.juno1"
11:48AM INF asserting crisis invariants inv=1/12 module=x/crisis name=distribution/nonnegative-outstanding
11:48AM INF asserting crisis invariants inv=2/12 module=x/crisis name=distribution/can-withdraw
11:48AM INF asserting crisis invariants inv=3/12 module=x/crisis name=distribution/reference-count
11:48AM INF asserting crisis invariants inv=4/12 module=x/crisis name=distribution/module-account
11:48AM INF asserting crisis invariants inv=5/12 module=x/crisis name=bank/nonnegative-outstanding
11:48AM INF asserting crisis invariants inv=6/12 module=x/crisis name=bank/total-supply
11:48AM INF asserting crisis invariants inv=7/12 module=x/crisis name=transfer/total-escrow-per-denom
11:48AM INF asserting crisis invariants inv=8/12 module=x/crisis name=gov/module-account
11:48AM INF asserting crisis invariants inv=9/12 module=x/crisis name=staking/module-accounts
11:48AM INF asserting crisis invariants inv=10/12 module=x/crisis name=staking/nonnegative-power
11:48AM INF asserting crisis invariants inv=11/12 module=x/crisis name=staking/positive-delegation
11:48AM INF asserting crisis invariants inv=12/12 module=x/crisis name=staking/delegator-shares
11:48AM INF asserted all invariants duration=1.706959 height=5 module=x/crisis
{"app_hash":"","app_state":{"auth":{"accounts":[{"@type":"/cosmos.auth.v1beta1.ModuleAccount",
```
